src/main/java/htsjdk/variant/utils/VCFHeaderReader.java

```diff
@@ -0,0 +1,54 @@
+package htsjdk.variant.utils;
+
+import htsjdk.samtools.SamStreams;
+import htsjdk.samtools.seekablestream.SeekableStream;
+import htsjdk.samtools.util.BlockCompressedInputStream;
+import htsjdk.tribble.FeatureCodecHeader;
+import htsjdk.tribble.TribbleException;
+import htsjdk.tribble.readers.AsciiLineReader;
+import htsjdk.tribble.readers.AsciiLineReaderIterator;
+import htsjdk.tribble.readers.PositionalBufferedStream;
+import htsjdk.variant.bcf2.BCF2Codec;
+import htsjdk.variant.vcf.VCFCodec;
+import htsjdk.variant.vcf.VCFHeader;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.zip.GZIPInputStream;
+
+/**
+ * Utility class to read a VCF header without being told beforehand whether the input is VCF or BCF.
+ */
+public final class VCFHeaderReader {
+    /**
+     * Read a VCF header from a stream assuming first that it is a VCF file (possibly gzip or block compressed),
+     * and falling back to trying BCF if reading VCF fails.
+     *
+     * @param in the stream to read the header from
+     * @return the VCF header read from the stream
+     * @throws IOException if no VCF header is found
+     */
+    public static VCFHeader readHeaderFrom(final SeekableStream in) throws IOException {
+        FeatureCodecHeader headerCodec;
+        final long initialPos = in.position();
+        try {
+            BufferedInputStream bis = new BufferedInputStream(in);
+            // despite the name, SamStreams.isGzippedSAMFile looks for any gzipped stream
+            InputStream is = SamStreams.isGzippedSAMFile(bis) ? new GZIPInputStream(bis) : bis;
```

Comment:
Does isGzippedSAMFile return true for a block compressed file?  It must or vcf.bgz wouldn't work here?  I think you should add a comment about that case here.

Reply:
Yes it does. Updated comment.